lib/check.js

```diff
@@ -0,0 +1,46 @@
+const Holidays = require("date-holidays");
+const scheduleMapping = ["monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday"]
+
+
+class Checker {
+    constructor(schedule, country) {
+        this.schedule = schedule || {}
+        if (country) {
+            this.holidays = new Holidays(country)
+        } else {
+            this.holidays = null
+        }
+    }
+
+    dayFromSchedule(date) {
+        return this.schedule[scheduleMapping[date.weekday - 1]];
+    }
+
+    checkWorkingDays(date) {
+        if (this.holidays && this.holidays.isHoliday(date)){
+            return false
+        }
+
+        if (this.dayFromSchedule(date)===undefined) {
+            return false
+        }
+
+        return true
+    }
+
+    checkWorkingHours(date) {
+        const day = this.dayFromSchedule(date)
+
+        if (day===undefined || day.startHour === undefined || day.endHour === undefined) {
+            return false
+        }
+
+        if (day.startHour > date.hour || date.hour >= day.endHour) {
+            return false
+        }
```

Comment:
This is easier to read.
```suggestion
        if (date.hour < day.startHour || date.hour >= day.endHour) {
            return false
        }
```

Reply:
Also the naming is little bit confusing. What is `day` and `date`? It could be `now` and `schedule_day`.